news_events/etl/utils.py

```diff
@@ -131,3 +132,69 @@ def parse_date(text_date: str) -> datetime:
         except:  # noqa: E722
             logging.exception("unparsable date received - ignoring '%s'", text_date)
     return dt_utc
+
+
+def convert_to_utc(dt: datetime, known_tz: str) -> datetime:
+    """
+    Convert a datetime object to UTC timezone. If its
+    orignal timezone is not known, assume it is in US/Eastern.
+
+    Args:
+        dt (datetime): The datetime object to convert
+        known_tz (str): The timezone string if known
+
+    Returns:
+        datetime: The datetime object in UTC timezone
+    """
+    if not dt:
+        return None
+    if not known_tz:
+        # Assume it is in US/Eastern where MIT is
+        dt = dt.replace(tzinfo=ZoneInfo("US/Eastern"))
+    return dt.astimezone(UTC)
+
+
+def parse_date_time_range(
+    start_date_str: str, end_date_str: str, time_range_str: str
+) -> tuple[datetime, datetime]:
+    """
+    Attempt to parse the time range from the MITPE events API.
+    The field might not actually contain a time or range.
+
+    Args:
+        start_date_str (str): start date string
+        end_date_str (str): end date string
+        time_range (str): time range string
+
+    Returns:
+        tuple: start and end datetimes
+
+    """
+    start_time, start_ampm, end_time, end_ampm, tz = "", "", "", "", ""
+    time_regex = re.compile(
+        r"(\d{1,2})(:\d{2})?\s*(am|pm)?\s*-?\s*(\d{1,2})(:?\d{2})?\s*(am|pm)?\s*([A-Za-z]{2,3})?",
```

Comment:
The regex is really complicated. Could it be broken up into multiple steps? Like one step to break the time string into two in the case where there is both a start and end date and a seperate step to extract pm/am. As it is i think it's hard to troubleshoot or understand what the regex is doing

Reply:
Split up the regex and also split up the function.  Added more inline comments and more unit test cases